src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs

```diff
@@ -110,6 +110,22 @@ public static float Dot(Vector2 value1, Vector2 value2)
                    value1.Y * value2.Y;
         }
 
+        /// <summary>
+        /// Returns the z-value of the cross product of two vectors.
+        /// Since the Vector2 is in the x-y plane, a 3D cross product
+        /// only produces the z-value
+        /// </summary>
+        /// <param name="value1">The first vector.</param>
+        /// <param name="value2">The second vector.</param>
+        /// <returns>The value of the z-coordinate from the cross product.</returns>
+        [JitIntrinsic]
```

Comment:
This would need to be `[Intrinsic]`. It may be better to not mark it as such until after we can make it actually intrinsic in the JIT though.

If https://github.com/dotnet/corefx/issues/36379 goes through API review and is approved, we can add the support in managed code instead of adding new intrinsic support for this to the JIT, which may be more maintainable.